### DISCUSSION TITLE
Could qdrant support on Android?

### DISCUSSION THREAD

**Discussion by KimmyYang** [qdrant/qdrant#2625] [General]
**Is your feature request related to a problem? Please describe.**
There's JAVA API(https://github.com/metaloom/qdrant-java-client) run for qdrant-client, but there's no qdrant-server can be ran on Android.

**Describe the solution you'd like**
Provides a way to build arm64-v8a qdrant-server binary for Android. 
The qdrant-server could launch `localhost:port` with using gRPC, the database can be located in App's local folder.

**Comment by agourlay** [comment]
Thanks for the idea :+1: 

Do you have experience building Rust project for Android?

We use [upload-rust-binary-action](https://github.com/taiki-e/upload-rust-binary-action) to generate executable for various platforms in https://github.com/qdrant/qdrant/blob/master/.github/workflows/release-artifacts.yml#L1

A good first step would to check if the current infrastructure could work for this use case.

**Reply by harshakhmk** [reply]
Through this package, is it possible to generate for Andriod platform

**Reply by agourlay** [reply]
No one has done the research yet.

If you are interested in this, I recommend you give it a try :)

**Reply by harshakhmk** [reply]
Hi, I will give it a try, but inorder to use that package, it is asking to add this code to .yml, but there seems to be no yml file in qdrant code
```
              - name: Build and upload Rust binary to GitHub Releases
                uses: taiki-e/upload-rust-binary-action@v1.18.0
            
```
Mentioned here https://github.com/marketplace/actions/build-and-upload-rust-binary-to-github-releases

**Reply by agourlay** [reply]
Our workflows are defined here https://github.com/qdrant/qdrant/tree/master/.github/workflows

**Reply by harshakhmk** [reply]
Is cross compilation possible for Android, if yes could you mention the steps

**Reply by agourlay** [reply]
I have no idea if it is possible so I will let other lurkers reply :)

**Reply by harshakhmk** [reply]
@generall @timvisee , could you kindly comment on this issue

**Reply by generall** [reply]
we have no experience with Android, it is not our target platform. Contributions are welcome

**Comment by KimmyYang** [comment]
@agourlay Thanks for the information.

Another request feature, could qdrant provide a way to build it as a library(ie. .so) instead of executable binary?

**Comment by agourlay** [comment]
> Another request feature, could qdrant provide a way to build it as a library(ie. .so) instead of executable binary?

I believing this has been a recurring demand, please create a new issue to be able to track it individually :+1:

**Reply by harshakhmk** [reply]
Hi is the . so Support for Qdrant Source code available to use it on Andriod

**Reply by agourlay** [reply]
I don't think we support the Android platform.
